examples/examples/error.rs

```diff
@@ -0,0 +1,68 @@
+#![deny(rust_2018_idioms)]
+use std::error::Error;
+use std::fmt;
+use tracing_error::{ErrorLayer, SpanTrace};
+use tracing_subscriber::{fmt::Layer as FmtLayer, prelude::*, registry::Registry};
+
+#[tracing::instrument]
+fn do_something(foo: &str) -> Result<&'static str, impl Error + Send + Sync + 'static> {
```

Comment:
Nit: Can the error be a type alias instead? It took me a few seconds to parse this signature. For instance, I think I'd prefer something like:

```rust
type Error = Box<dyn std::Error + Send + Sync + 'static>;

#[tracing::instrument]
fn do_something(foo: &str) -> Result<&'static str, Error> { ...}
```